api/admin/controller/__init__.py

```diff
@@ -293,6 +299,7 @@ def get_csrf_token(self):
 
     def generate_csrf_token(self):
         """Generate a random CSRF token."""
+        # NOTE: We don't use random_string here -- maybe we could.
         return base64.b64encode(os.urandom(24))
```

Comment:
I didn't change this because random_string(24) is going to encode the random string as hex digits, which will give a significantly longer string than base64-encoding.  I don't know if that distinction is important.

Reply:
This token ends up in a cookie and also passed down to the front-end code. As long as it's not exceeding the maximum cookie size I think it would be fine.